pkg/encoding/encoding.go

```diff
@@ -63,15 +63,19 @@ func Marshal(format string, in interface{}) (data []byte, err error) {
 		return nil, newUnsupportedFormatError(format)
 	}
 
-	return data, errors.Wrapf(err, "error marshaling to %s", format)
+	if err != nil {
+		return data, fmt.Errorf("error marshaling to %s: %w", format, err)
```

Comment:
Let's only return the first argument when there isn't an error
```suggestion
		return nil, fmt.Errorf("error marshaling to %s: %w", format, err)
```